taskcluster/translations_taskgraph/transforms/cached_tasks.py

```diff
@@ -74,3 +88,48 @@ def add_cache(config, jobs):
         }
 
         yield job
+
+
+@transforms.add
+def cache_task(config, tasks):
+    if taskgraph.fast:
+        for task in tasks:
+            yield task
+        return
+
+    digests = {}
+    for task in config.kind_dependencies_tasks.values():
+        if "cached_task" in task.attributes:
+            digests[task.label] = format_task_digest(task.attributes["cached_task"])
+
+    for task in order_tasks(config, tasks):
+        cache = task.pop("cache", None)
+        if cache is None:
+            yield task
+            continue
+
+        dependency_digests = []
+        for p in task.get("dependencies", {}).values():
+            # Here in Translations, we explicit do _not_ invalidate cached
+            # training jobs when non-training jobs are invalidated.
+            if any([kind in p for kind in DONT_INVALIDATE_KINDS]):
```

Comment:
Mm this might be nice on some other projects. But, here it's really needed.

Reply:
Hm...this makes me wonder if this is upstreamable as well...